shared/Microsoft.Extensions.Certificates.Generation.Sources/CertificateManager.cs

```diff
@@ -0,0 +1,414 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+
+namespace Microsoft.Extensions.Certificates.Generation
+{
+    internal class CertificateManager
+    {
+        public const string AspNetHttpsOid = "1.3.6.1.4.1.311.84.1.1";
+        public const string AspNetHttpsOidFriendlyName = "Asp.Net Core HTTPS development certificate";
+
+        public const string AspNetIdentityOid = "1.3.6.1.4.1.311.84.1.2";
+        public const string AspNetIdentityOidFriendlyName = "Asp.Net Core Identity Json Web Token signing development certificate";
+
+        private const string ServerAuthenticationEnhancedKeyUsageOid = "1.3.6.1.5.5.7.3.1";
+        private const string ServerAuthenticationEnhancedKeyUsageOidFriendlyName = "Server Authentication";
+
+        private const string LocalhostHttpsDnsName = "localhost";
+        private const string LocalhostHttpsDistinguishedName = "CN=" + LocalhostHttpsDnsName;
+
+        private const string IdentityDistinguishedName = "CN=Microsoft.AspNetCore.Identity.Signing";
+
+        public const int RSAMinimumKeySizeInBits = 2048;
+
+        public IList<X509Certificate2> ListCertificates(
+            CertificatePurpose purpose,
+            StoreName storeName,
+            StoreLocation location,
+            bool isValid)
+        {
+            var certificates = new List<X509Certificate2>();
```

Comment:
Unless I'm missing something, it seems like you could simplify by calling. Not sure why you need this list yet.
```
var certificates = store.Certificates.OfType<X509Certificate2>().ToList();
```